openmcyclus/depletion.py

```diff
@@ -3,156 +3,192 @@
+            list of the fresh fuel compositions present in the core
+            at the calling of the transmute function.
+        path: str
+            path of directory holding the files for/from OpenMC
+
+        Returns:
+        --------
+        material_ids: list of strs
+            material id numbers for the OpenMC model
+
+        Outputs:
+        --------
+        materials.xml: file
+            updated XML for OpenMC with new compositions
+
+        '''
+        openmc_materials = ET.parse(str(path + "materials.xml"))
+        openmc_root = openmc_materials.getroot()
+        material_ids = []
+
+        for child in openmc_root:
+            if '_' in child.attrib['name']:
+                underscore_index = child.attrib['name'].index('_')
+                assembly_number = child.attrib['name'][underscore_index + 1:]
+                material_ids.append(child.attrib['id'])
+                if child.attrib['name'][:underscore_index] == 'assembly':
+                    for material in child.findall('nuclide'):
+                        child.remove(material)
+                    new_comp = comp_list[int(assembly_number) - 1]
+                    for nuclide in new_comp:
+                        Z = math.floor(nuclide / int(1e7))
+                        A = math.floor((nuclide - Z * int(1e7)) / int(1e4))
+                        m = nuclide - Z * int(1e7) - A * int(1e4)
+                        nucname = openmc.data.gnd_name(Z, A, m)
+                        new_nuclide = f"""<nuclide wo="{str(new_comp[nuclide]*100)}" name="{nucname}" />"""
+                        new_nuclide_xml = ET.fromstring(new_nuclide)
+                        child.insert(1, new_nuclide_xml)
+        ET.indent(openmc_root)
+        openmc_materials.write(str(path + "materials.xml"))
+        return material_ids
+
+    def read_microxs(self, path):
         '''
         Reads .csv file with microscopic cross sections. The
         csv file is assumed to be named "micro_xs.csv". This will need to
         be relative to Cyclus input file location
 
         Parameters:
         -----------
+        path: str
+            path of directory holding the files for/from OpenMC        
 
         Returns:
         --------
         microxs: object
             microscopic cross section data
         '''
-        microxs = od.MicroXS.from_csv(self.path / "micro_xs.csv")
+        microxs = od.MicroXS.from_csv(str(path + "micro_xs.csv"))
         return microxs
 
-    def run_depletion(self):
+    def run_depletion(self, path):
         '''
         Run the IndependentOperator class in OpenMC to perform
         transport-independent depletion.
 
         Parameters:
-        -----------
+        -----------        
+        path: str
+            path of directory holding the files for/from OpenMC
 
         Outputs:
         --------
         depletion_results.h5: database
             HDF5 data base with the results of the depletion simulation
```

Comment:
Since the function returns `None` I would simply leave out the second part of this and describe it in the main part of the docstring. E.g. "writes to an HDF5 database called depletion_results.h5 with the simulation results."

```suggestion
        path : str
            path of directory holding the files for/from OpenMC.
```